addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -712,5 +763,100 @@ odoo.define('website.s_website_form', function (require) {
         _onFieldInput() {
             this._updateFieldsVisibility();
         },
+        /**
+         * Called when files are uploaded: updates the button text content,
+         * displays the file blocks (containing the files name and a cross to
+         * delete them) and manages the files.
+         *
+         * @private
+         * @param {Event} ev
+         */
+        _onFileChange(ev) {
+            const fileInputEl = ev.currentTarget;
+            const fieldEl = fileInputEl.closest('.s_website_form_field');
+            const uploadedFiles = fileInputEl.files;
+            const addFilesButtonEl = fieldEl.querySelector('.o_add_files_button');
+
+            // The zone where the file blocks are displayed.
+            let filesZoneEl = fieldEl.querySelector('.o_files_zone');
+            // Update the button text content.
+            if (!addFilesButtonEl) {
+                this._createAddFilesButton(fileInputEl);
+            }
+
+            // Create a list to keep track of the files.
+            if (!fileInputEl.fileList) {
+                fileInputEl.fileList = new DataTransfer();
+            }
+
+            // If only one file can be uploaded, delete the previous file.
+            if (!fileInputEl.getAttribute('multiple') && uploadedFiles.length > 0) {
+                fileInputEl.fileList = new DataTransfer();
+                const fileBlockEl = fieldEl.querySelector('.o_file_block');
+                if (fileBlockEl) {
+                    fileBlockEl.remove();
+                }
+            }
+
+            // Add the uploaded files if they are not already there.
+            for (const newFile of uploadedFiles) {
+                let isFileAlreadyUploaded = false;
+                for (const file of fileInputEl.fileList.files) {
+                    if (newFile.name === file.name && newFile.size === file.size && newFile.type === file.type) {
+                        isFileAlreadyUploaded = true;
+                        break;
+                    }
+                }
+                if (!isFileAlreadyUploaded) {
+                    fileInputEl.fileList.items.add(newFile);
+                    const fileDetails = {name: newFile.name, size: newFile.size, type: newFile.type};
+                    this._createFileBlock(fileDetails, filesZoneEl);
+                }
```

Comment:
I think you could do this in a more appropriate way (without using the `isFileAlreadyUploaded` variable) and in less lines with the `some();` method.
```js
if (fileInputEl.fileList.files.some(file => newFile.name === file.name && newFile.size === file.size && newFile.type === file.type)) {
    ....
}